runtime/runtime/src/lib.rs

```diff
@@ -452,6 +455,20 @@ impl Runtime {
             }
         };
 
+        if result.result.is_err() && result.burnt_gas_reward > 0 {
+            // The transaction has failed but we still need to provide contract rewards.
```

Comment:
should this just be in the Err(e) arm of match above?

Reply:
Agreed. You can use guards if needed https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html